### DISCUSSION TITLE
API for "Re-request a voltage change on the fly"?

### DISCUSSION THREAD

**Discussion by cubemike** [Ralim/usb-pd#15] [Q&A]
Hi,

I'm wondering how to change the voltage after the initial negotiation. Forgive me if there's already a well written guide, I haven't found it browsing the repo. But just browsing the code, it seems that one way to do it would be to run PolicyEngine::notify(PolicyEngine::Notifications::NEW_POWER) in user code to have the state machine rerun pdbs_dpm_evaluate_capability.

However, notify() and the Notifications enum are private. Are they intended to be private, and I'm not seeing the correct way to change voltage? Or should they in fact be public?

Thanks,
-Mike

**Comment by Ralim** [comment]
Ah oops.

There was intentioned to be a function call that's public that would trigger that exact notification.

Happy for a PR or I can get it in there for you within a day or two.
